Source/Particles/LaserParticleContainer.cpp

```diff
@@ -236,15 +237,22 @@ LaserParticleContainer::UpdateContinuousInjectionPosition (Real dt)
 void
 LaserParticleContainer::InitData ()
 {
+    if (!m_enabled) return;
+
     // Call InitData on max level to inject one laser particle per
     // finest cell.
     InitData(maxLevel());
+
+    if(!do_continuous_injection && (TotalNumberOfParticles() == 0)){
```

Comment:
That might be slightly more obvious, but the current version is also ok:
```suggestion
    if(!do_continuous_injection && (this->TotalNumberOfParticles() == 0)){
```